### ISSUE TITLE
Session unserialization does not work with shared object between Serializable objects with new PHP 7.4

### ISSUE THREAD

**Issue by syberon** [zendframework/zend-session#126]
After updating PHP to version 7.4 authorization stop working. On next page loading after success authorization i get this error:

```
Notice: unserialize(): Error at offset 43 of 224 bytes in C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-stdlib\src\ArrayObject.php on line 409

Notice: Trying to access array offset on value of type bool in C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-stdlib\src\ArrayObject.php on line 412

Notice: Trying to access array offset on value of type bool in C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-stdlib\src\ArrayObject.php on line 413

Warning: session_start(): Failed to decode session object. Session has been destroyed in C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-session\src\SessionManager.php on line 138

Fatal error: Uncaught Zend\Stdlib\Exception\InvalidArgumentException: Passed variable is not an array or object, using empty array instead in C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-stdlib\src\ArrayObject.php:184 Stack trace: #0 C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-stdlib\src\ArrayObject.php(413): Zend\Stdlib\ArrayObject->exchangeArray(NULL) zendframework/zend-authentication#1 [internal function]: Zend\Stdlib\ArrayObject->unserialize('a:4:{s:7:"stora...') zendframework/zend-authentication#2 C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-session\src\SessionManager.php(138): session_start() zendframework/zend-authentication#3 C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-session\src\AbstractContainer.php(81): Zend\Session\SessionManager->start() zendframework/zend-authentication#4 C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-authentication\src\Storage\Session.php(63): Zend\Session\AbstractContainer->__construct('Zend_Auth', NULL) zendframework/zend-authentication#5 C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-authentication\src\AuthenticationService.php(78): Zend\Authentication\Storage\Session->__construct() zendframework/zend-authentication#6 C:\SERVER\project in C:\SERVER\projects\zf3\mt4\vendor\zendframework\zend-stdlib\src\ArrayObject.php on line 184
```

After this error PHP destroys current session with authorized storage data and creates new.
I found that error is appeared when PHP trying to unserialize data stored in session that related to 'Zend_Auth' namespace:

```
Zend_Auth|C:23:"Zend\Stdlib\ArrayObject":224:{a:4:{s:7:"storage";a:1:{s:7:"storage";r:20;}s:4:"flag";i:2;s:13:"iteratorClass";s:13:"ArrayIterator";s:19:"protectedProperties";a:4:{i:0;s:7:"storage";i:1;s:4:"flag";i:2;s:13:"iteratorClass";i:3;s:19:"protectedProperties";}}}

```

Error says that `Error at offset 43 of 224 bytes` and at this position is `r:20` (reference to object in another session namespace)

**Comment by syberon** [comment]
> It is not something that we can fix. Php bug report must be opened if there is none yet.

I'm already opened a report in bugs.php.net 2 days ago, but there is still no response.

**Comment by Xerkus** [comment]
Php bugtracker link for reference https://bugs.php.net/bug.php?id=79031

**Comment by syberon** [comment]
Thank for help with problem description. I hope developers will fix it asap ) I think this problem is related to last serialize improvement (introduces the new magic methods) in PHP 7.4.
The quick workaround to fix Zend_Authentication to work with 7.4 it to remove Serializable implementation from ArrayObject class declaration (not good solution but as temporary...)

**Comment by michalbundyra** [comment]
Closing, as it is a bug in PHP 7.4 not in the library itself.
